src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -2179,11 +2531,116 @@ class CgroupUtils(object):
                 else:
                     pbs.logmsg(pbs.EVENT_DEBUG4, "%s: Unknown subsystem %s" %
                                (caller_name(), key))
-                    raise(CgroupConfigError, "Unknown subsystem: %s" % (key))
+                    raise CgroupConfigError("Unknown subsystem: %s" % key)
         pbs.logmsg(pbs.EVENT_DEBUG4, "%s: Returning %s" %
                    (caller_name(), str(assigned)))
         return assigned
 
+    def _get_systemd_version(self):
+        """
+        Return an integer reflecting the systemd version, zero for no systemd
+        """
+        pbs.logmsg(pbs.EVENT_DEBUG4, "%s: Method called" % caller_name())
+        try:
+            process = subprocess.Popen(['systemctl', '--version'], shell=False,
+                                       stdout=subprocess.PIPE,
+                                       stderr=subprocess.PIPE)
+            out, err = process.communicate()
+            ver = int(re.sub(r'systemd ([0-9]+)', r'\1', out.splitlines()[0]))
+        except KeyboardInterrupt:
+            raise
+        except:
```

Comment:
In this case, I really don't care what the error was or whether it came from the Popen() call or something else. We just want to return a version for systemd. It could be argued that goto statements in C are a bad idea, but they are useful in some limited circumstances. I think the blanket except is a similar case.

Reply:
I'm not sure I agree the two are equatable.  You are right, for the most part goto statements in C are a bad idea.  There are a very small number of cases where they are OK (usually to increase readability).  I dislike them, but if you really want the blanket except, I won't complain any further.